src/DefaultHeadersRequest.jl

```diff
@@ -0,0 +1,62 @@
+module DefaultHeadersRequest
+
+export defaultheaderslayer, setuseragent!
+
+using ..Messages
+using ..IOExtras
+import ..Form, ..content_type
+
+"""
+    defaultheaderslayer(req) -> Response
+
+Sets default expected headers.
+"""
+function defaultheaderslayer(handler)
+    return function(req; iofunction=nothing, kw...)
+        headers = req.headers
+        if isempty(req.url.port) ||
+            (req.url.scheme == "http" && req.url.port == "80") ||
+            (req.url.scheme == "https" && req.url.port == "443")
+            hostheader = req.url.host
+        else
+            hostheader = req.url.host * ":" * req.url.port
+        end
+        defaultheader!(headers, "Host" => hostheader)
+        defaultheader!(headers, "Accept" => "*/*")
+        if USER_AGENT[] !== nothing
+            defaultheader!(headers, "User-Agent" => USER_AGENT[])
+        end
+
+        if !hasheader(headers, "Content-Length") &&
+        !hasheader(headers, "Transfer-Encoding") &&
+        !hasheader(headers, "Upgrade")
+            l = nbytes(req.body)
+            if l !== nothing
```

Comment:
```suggestion
            if nbytes(req.body) !== nothing
```

Reply:
We use the `l` variable in the next line; but this is probably due to being a really poor variable name more than anything.